arcade/application.py

```diff
@@ -930,6 +930,19 @@ def on_mouse_leave(self, x: int, y: int):
         """
         pass
 
+    def save_screenshot(self, location: Optional[str] = None) -> Path:
+
+        img = self.ctx.get_framebuffer_image(self.ctx.screen)
+        if not location:
+            output_dir = Path().parent.absolute()
+        else:
+            output_dir = Path(location).absolute()
+
+        filename = f"{self.caption.lower().replace(' ', '_')}_{time.monotonic_ns()}.png"
+        full_file_path = output_dir / filename
+        img.save(full_file_path, 'PNG')
```

Comment:
I'm not sure on this behavior. This is hard-locking the path and format.

Reply:
So what do we want to do here?
My thinking based on the discussion was that only input for the time being would be a path to store the files.  Then we need to auto-gen a filename based on some format and save it in that path.  I figured that the window title and timestamp would do the trick.  We can make it fancier if you want.  We can make params with sane defaults for filename generation for example, but honestly whenever I use screenshot functionality I would not expect to control it at that level.
Thoughts?

Reply:
I guess we could just make it save the image with whatever the user passes in.  They control the file and path creation and just pass in the full deal.  Is that more what you were thinking